test/gurka/test_ferry_connections.cc

```diff
@@ -3,12 +3,13 @@
 
 using namespace valhalla;
 
-baldr::RoadClass
+std::pair<baldr::RoadClass, baldr::RoadClass>
 ReclassifyFerryConnectionEdge(std::map<std::string, std::string> const& way_description) {
   constexpr double gridsize_metres = 1000;
 
   const std::string ascii_map = R"(
           A--B--b--C----D--E
+          F--G--g--H----I--J
```

Comment:
Adding a parallel test case ensuring that the behavior for rail ferries is the same as for ferries. The only difference in these two test cases is that `HI` is a rail ferry and `CD` is a ferry.